### ISSUE TITLE
Yolo learning rate in logs not changing

### ISSUE THREAD

**Issue by deepakcrk** [pjreddie/darknet#202]
In my YOLO config I have the following:

```
learning_rate=0.001
steps=-1,100,1000,2000
scales=.1,10,.1,.1
```

But the learning rate value in the logs is not changing:

```
3336: 29.154839, 30.304903 avg, 0.001000 rate, 3.655274 seconds, 213504 images
3337: 28.832424, 30.157656 avg, 0.001000 rate, 3.600136 seconds, 213568 images
3338: 31.017210, 30.243610 avg, 0.001000 rate, 3.688117 seconds, 213632 images
3339: 30.417107, 30.260960 avg, 0.001000 rate, 3.577419 seconds, 213696 images
3340: 30.913982, 30.326262 avg, 0.001000 rate, 3.553818 seconds, 213760 images
3341: 30.681067, 30.361742 avg, 0.001000 rate, 3.689000 seconds, 213824 images
3342: 30.039232, 30.329491 avg, 0.001000 rate, 3.621938 seconds, 213888 images
```

Why is it not decreasing the value after the steps specified in the cfg file? The learning rate in logs is always `0.001000`, which is my initial learning rate.

**Comment by dtmoodie** [comment]
What do you have your policy set to?

**Comment by deepakcrk** [comment]
steps

**Comment by deepakcrk** [comment]
steps

**Comment by deepakcrk** [comment]
I am using yolo.2.cfg

**Comment by ZIWEILI** [comment]
You can try this:  go to your src/network.c, find the function `float get_current_rate(network net)` , for`case STEPS:`, change `if(net.steps[i] > batch_num) ` to `if(net.steps[i] > (int)batch_num) `.
